File: t/CS/ImageProcessingDemo/UI/Command/HistogramEqualizeDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools.ImageProcessing.Color;

namespace ImageProcessingDemo
{
   //This dialog is used to allow the user to specify values which
   //will be used for the HistogramEqualizeCommand

   public partial class HistogramEqualizeDialog : Form
   {
      private static bool _firstTimer = true;
      private static HistogramEqualizeType _initialColorSpace;
      private HistogramEqualizeCommand _HistogramEqualizeCommand;

      public HistogramEqualizeType ColorSpace;

      public HistogramEqualizeDialog( )
      {
         InitializeComponent();
         _HistogramEqualizeCommand = new HistogramEqualizeCommand(); 
      }

      public HistogramEqualizeCommand HistogramEqualizeCommand
      {
         get
         {
            return _HistogramEqualizeCommand;
         }
         set
         {
            _HistogramEqualizeCommand = value; 
         }

      }
      private void HistogramEqualizeDialog_Load(object sender, System.EventArgs e)
      {
         if(_firstTimer)
         {
            _firstTimer = false;
            _initialColorSpace = _HistogramEqualizeCommand.Type;
         }

         ColorSpace = _initialColorSpace;

         //Set command default values
         string[] names;
         names = Enum.GetNames(typeof(HistogramEqualizeType));
         foreach (string name in names)
         {
            if(name != "None")
            _cmbColorSpace.Items.Add(name);
         }
         _cmbColorSpace.SelectedIndex = _cmbColorSpace.Items.IndexOf(ColorSpace.ToString());
      }
      public HistogramEqualizeType TranslateType()
      {
         switch (_cmbColorSpace.SelectedIndex)
         {
            case 0:
               return HistogramEqualizeType.Rgb;
            case 1:
               return HistogramEqualizeType.Yuv; 
            case 2:
               return HistogramEqualizeType.Gray; 
            default:
               return HistogramEqualizeType.Rgb;
         }
      }
      private void _btnOk_Click(object sender, System.EventArgs e)
      {
         //Update command values
         ColorSpace = TranslateType();
         _HistogramEqualizeCommand.Type = ColorSpace;  
         _initialColorSpace = ColorSpace;
         this.DialogResult = DialogResult.OK;
      }

      private void _btnCancel_Click(object sender, EventArgs e)
      {
         this.DialogResult = DialogResult.Cancel;
      }
   }
}

File: t/VB/ImageProcessingDemo/UI/Command/HistogramEqualizeDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools.ImageProcessing.Color

Namespace ImageProcessingDemo
   'This dialog is used to allow the user to specify values which
   'will be used for the HistogramEqualizeCommand

   Partial Public Class HistogramEqualizeDialog : Inherits Form
      Private Shared _firstTimer As Boolean = True
      Private Shared _initialColorSpace As HistogramEqualizeType
      Private _HistogramEqualizeCommand As HistogramEqualizeCommand

      Public ColorSpace As HistogramEqualizeType

      Public Sub New()
         InitializeComponent()
         _HistogramEqualizeCommand = New HistogramEqualizeCommand()
      End Sub

      Public Property HistogramEqualizeCommand() As HistogramEqualizeCommand
         Get
            Return _HistogramEqualizeCommand
         End Get
         Set(ByVal value As HistogramEqualizeCommand)
            _HistogramEqualizeCommand = value
         End Set

      End Property
      Private Sub HistogramEqualizeDialog_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
         If _firstTimer Then
            _firstTimer = False
            _initialColorSpace = _HistogramEqualizeCommand.Type
         End If

         ColorSpace = _initialColorSpace

         'Set command default values
         Dim names As String()
         names = System.Enum.GetNames(GetType(HistogramEqualizeType))
         For Each name As String In names
            If name <> "None" Then
               _cmbColorSpace.Items.Add(name)
            End If
         Next name
         _cmbColorSpace.SelectedIndex = _cmbColorSpace.Items.IndexOf(ColorSpace.ToString())
      End Sub
      Public Function TranslateType() As HistogramEqualizeType
         Select Case _cmbColorSpace.SelectedIndex
            Case 0
               Return HistogramEqualizeType.Rgb
            Case 1
               Return HistogramEqualizeType.Yuv
            Case 2
               Return HistogramEqualizeType.Gray
            Case Else
               Return HistogramEqualizeType.Rgb
         End Select
      End Function
      Private Sub _btnOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles _btnOk.Click
         'Update command values
         ColorSpace = TranslateType()
         _HistogramEqualizeCommand.Type = ColorSpace
         _initialColorSpace = ColorSpace
         Me.DialogResult = DialogResult.OK
      End Sub

      Private Sub _btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnCancel.Click
         Me.DialogResult = DialogResult.Cancel
      End Sub
   End Class
End Namespace
